File: configuration/src/test/java/com/synopsys/integration/configuration/config/KeyUtilsTests.java
package com.synopsys.integration.configuration.config;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.synopsys.integration.configuration.util.KeyUtils;

public class KeyUtilsTests {
    @Test
    public void replacesUnderscores() {
        Assertions.assertEquals("key.dot.dot", KeyUtils.normalizeKey("key_dot_dot"));
    }

    @Test
    public void replacesCapitals() {
        Assertions.assertEquals("key.lower", KeyUtils.normalizeKey("KEY.LOWER"));
    }

    @Test
    public void normalizesEnvKey() {
        Assertions.assertEquals("env.key", KeyUtils.normalizeKey("ENV_KEY"));
    }
}

File: configuration/src/test/java/com/synopsys/integration/configuration/config/KeyUtilsTests.kt
package com.synopsys.integration.configuration.config

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

open class KeyUtilsTests {
    @Test
    fun replacesUnderscores() {
        Assertions.assertEquals("key.dot.dot", KeyUtils.normalizeKey("key_dot_dot"))
    }

    @Test
    fun replacesCapitals() {
        Assertions.assertEquals("key.lower", KeyUtils.normalizeKey("KEY.LOWER"))
    }

    @Test
    fun normalizesEnvKey() {
        Assertions.assertEquals("env.key", KeyUtils.normalizeKey("ENV_KEY"))
    }
}